File: sdks/java/src/main/java/com/boltprediction/sdk/model/FederatedPrivacyConfig.java
package com.boltprediction.sdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;
import java.util.Map;
import java.time.OffsetDateTime;

/** 隐私保护配置 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class FederatedPrivacyConfig {

    @JsonProperty("mechanism")
    private String mechanism;

    @JsonProperty("epsilon")
    private Double epsilon;

    @JsonProperty("delta")
    private Double delta;

    @JsonProperty("noise_scale")
    private Double noiseScale;

    @JsonProperty("clip_norm")
    private Double clipNorm;

    @JsonProperty("num_parties")
    private Integer numParties;

    @JsonProperty("secret_share_threshold")
    private Integer secretShareThreshold;

    public String getMechanism() {
        return mechanism;
    }

    public void setMechanism(String mechanism) {
        this.mechanism = mechanism;
    }

    public Double getEpsilon() {
        return epsilon;
    }

    public void setEpsilon(Double epsilon) {
        this.epsilon = epsilon;
    }

    public Double getDelta() {
        return delta;
    }

    public void setDelta(Double delta) {
        this.delta = delta;
    }

    public Double getNoiseScale() {
        return noiseScale;
    }

    public void setNoiseScale(Double noiseScale) {
        this.noiseScale = noiseScale;
    }

    public Double getClipNorm() {
        return clipNorm;
    }

    public void setClipNorm(Double clipNorm) {
        this.clipNorm = clipNorm;
    }

    public Integer getNumParties() {
        return numParties;
    }

    public void setNumParties(Integer numParties) {
        this.numParties = numParties;
    }

    public Integer getSecretShareThreshold() {
        return secretShareThreshold;
    }

    public void setSecretShareThreshold(Integer secretShareThreshold) {
        this.secretShareThreshold = secretShareThreshold;
    }

}

File: sdks/python/bolt_prediction_sdk/models/federated_privacy_config.py
"""
FederatedPrivacyConfig 模型
"""

from typing import Optional, Any, Dict, List
from datetime import datetime
from pydantic import Field

from .base import SDKBaseModel


class FederatedPrivacyConfig(SDKBaseModel):
    """隐私保护配置"""

    mechanism: Optional[str] = Field(description="隐私机制: none/dp/secagg/combined", default='none')
    epsilon: Optional[float] = Field(description="差分隐私epsilon", default=1.0)
    delta: Optional[float] = Field(description="差分隐私delta", default=1e-05)
    noise_scale: Optional[float] = Field(description="噪声缩放系数", default=0.1)
    clip_norm: Optional[float] = Field(description="梯度裁剪范数", default=1.0)
    num_parties: Optional[int] = Field(description="安全聚合参与方数量", default=3)
    secret_share_threshold: Optional[int] = Field(description="秘密共享阈值", default=2)
